### PR TITLE
cabal-install: Depends on zlib for Linuxbrew

### PR DESCRIPTION
**Pull request by lordqwerty**
- [X] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by lordqwerty** [comment]
How strange it works on CircleCI and not Travis. Error:
```
==> brew test cabal-install --verbose
Testing cabal-install
==> /home/linuxbrew/.linuxbrew/Cellar/cabal-install/1.24.0.2_4/bin/cabal --config-file=/tmp/cabal-install-test-20170402-4624-3yn40o/config info cabal
.
Last 150 lines from /home/linuxbrew/.cache/Homebrew/Logs/cabal-install/test.01.cabal:
2017-04-02 22:25:49 +0000
/home/linuxbrew/.linuxbrew/Cellar/cabal-install/1.24.0.2_4/bin/cabal
--config-file=/tmp/cabal-install-test-20170402-4624-3yn40o/config
info
cabal
/home/linuxbrew/.linuxbrew/Cellar/cabal-install/1.24.0.2_4/bin/cabal: error while loading shared libraries: libz.so.1: cannot open shared object file: No such file or directory
Error: cabal-install: failed
Failed executing: /home/linuxbrew/.linuxbrew/Cellar/cabal-install/1.24.0.2_4/bin/cabal --config-file=/tmp/cabal-install-test-20170402-4624-3yn40o/config info cabal
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1836:in `block in system'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1774:in `open'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1774:in `system'
/home/linuxbrew/.linuxbrew/Library/Taps/homebrew/homebrew-core/Formula/cabal-install.rb:29:in `block in <class:CabalInstall>'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1673:in `block (2 levels) in run_test'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:884:in `with_logging'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1672:in `block in run_test'
/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:14:in `block in mktemp'
/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `block in run'
/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `chdir'
/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `run'
/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:13:in `mktemp'
/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1666:in `run_test'
/home/linuxbrew/.linuxbrew/Library/Homebrew/test.rb:33:in `block in <main>'
/home/travis/.rvm/rubies/ruby-2.0.0-p648-clang/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/home/linuxbrew/.linuxbrew/Library/Homebrew/test.rb:32:in `<main>'
==> FAILED
```

**Comment by lordqwerty** [comment]
Linux bottle builds!

File: Formula/cabal-install.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class CabalInstall < Formula
   homepage "https://www.haskell.org/cabal/"
   url "https://hackage.haskell.org/package/cabal-install-1.24.0.2/cabal-install-1.24.0.2.tar.gz"
   sha256 "2ac8819238a0e57fff9c3c857e97b8705b1b5fef2e46cd2829e85d96e2a00fe0"
-  revision 3
+  revision 4
```
Comment (sjackman) [inline Formula/cabal-install.rb]: Please revert this chunk. It's not necessary to bump `revision`.

File: Formula/cabal-install.rb [hunk 2]
```diff
@@ -13,6 +13,7 @@ class CabalInstall < Formula
     sha256 "f58112b5b74e0056013daf2ac47b0f20393904557ff4f994ef5361e0ce3db343" => :yosemite
   end
 
+  depends_on "zlib" => :build unless OS.mac?
```
Comment (sjackman) [inline Formula/cabal-install.rb]: Try instead
```ruby
depends_on "zlib" unless OS.mac?
```
unless you know `zlib` is not a run-time dependency, which would be unusual.

**Comment by sjackman** [comment]
Nice! Thanks for your contribution to Linuxbrew, Jorden. For future contributions, the preferred commit log message is
```
cabal-install: Depends on zlib for Linuxbrew
```

**Pull request closed without merging**